gemrb/core/Map.cpp

```diff
@@ -2392,16 +2392,19 @@ void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 {
 	size_t pl = SongList[SongType];
 	const ieVariable* poi = &core->GetMusicPlaylist(pl);
+	Game* game = core->GetGame();
 
 	// Some subareas don't have their own songlist and GemRB would fail to play
 	// the expected music then. Currently this is only observed for the battle
 	// music in BG1, e.g. AR2607 (intro candlekeep ambush south).
-	// In this case GemRB falls back to the music referenced directly by SongType.
-	// it's not the correct music, perhaps it needs the one from the master area
-	// it would match for ar2607 and ar2600, but very annoying (see GetMasterArea)
+	// In this case GemRB now plays the music from the masterarea's songlist,
+	// or falls back to the music referenced directly by SongType if the masterarea
+	// can't be determined (almost always incorrect, but better than nothing).
```

Comment:
which areas does it not work for or what does the incorrectness comment refer to?

Reply:
The incorrectness comment refers to the fallback if the masterarea can't be determined (`GetMap` returning `nullptr` for whatever reason). Here we have `SongType == SONG_BATTLE == 3`, so we alway access the same music `GetMusicPlaylist(3)` in this fallback case. For BG1 it is `CNITE.mus`. I bet this is almost always not the correct battle music ;)

But yeah, I should probably restructure the comment a little bit to clarify it.

Reply:
Yes, that GetMusicPlaylist call will always be bad. Our music.2da is most probably fine, so we should choose one of 19-29 instead in this fallback. That's also obviously not correct for other games.

Reply:
Indeed that's a good idea to just use a proper battlemusic in this case then. Defaults now to 19 for last resort fallback.